<!-- launch file to start the main RTAB-Map node -->
<launch>
	<!-- static transform publisher from quadrotor to camera frame -->
	<node pkg = "tf" type = "static_transform_publisher" name="camera_transform_publisher" args = "0.245 0 0.03 -0.5 0.5 -0.5 0.5 quadrotor camera 50"/>

	<!-- static transform publisher from camera to depth-camera frame -->
	<node pkg = "tf" type = "static_transform_publisher" name="depth_transform_publisher" args = "0 -0.046 0.004 0 0 0 camera depth 50"/>
	
	<!-- start the rtabmap node -->
	<group ns="rtabmap">
		<node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start --udebug">   
			<param name="RGBD/OptimizeIterations" type="string" value="50"/>
			<param name="odom_frame_id" type="string" value="odom"/>
			<param name="subscribe_depth" type="bool" value="true"/>
			<param name="frame_id" type="string" value="quadrotor"/>
			<param name="approx_sync" type="bool" value="true"/>
			<param name="cloud_max_depth" type="double" value="3.0"/>

			<remap from="rgb/image" to="/realsense/camera/color/image_raw"/>
			<remap from="rgb/camera_info" to="/realsense/camera/color/camera_info"/>
			<remap from="depth/image" to="/depth_registered/image_rect"/>

			<param name="RGBD/AngularUpdate" type="string" value="0.01"/>
			<param name="RGBD/LinearUpdate" type="string" value="0.01"/>
			<param name="Rtabmap/TimeThr" type="string" value="700"/>
			<param name="Mem/RehearsalSimilarity" type="string" value="0.45"/>
			<param name="RGBD/OptimizeFromGraphEnd" type="string" value="true"/>		
			<param name="queue_size" type="int" value="30"/>
		</node>
	</group>

	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find realsense_gazebo_plugin)/rviz/keyboard_mapping.rviz" required="true" />

</launch>
